recipes/aws-cdi-sdk/all/conanfile.py

```diff
@@ -0,0 +1,93 @@
+from enum import auto
+from conans import ConanFile, AutoToolsBuildEnvironment, tools, CMake
+from conans.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.35.0"
+
+class AwsCdiSdkConan(ConanFile):
+    name = "aws-cdi-sdk"
+    description = "AWS Cloud Digital Interface (CDI) SDK"
+    topics = ("aws", "communication", "framework", "service")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aws/aws-cdi-sdk"
+    license = "BSD-2-Clause"
+    settings = "os", "arch", "compiler", "build_type"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    generators = "cmake", "cmake_find_package"
+    requires = "aws-libfabric/1.9.1amzncdi1.0", "aws-sdk-cpp/1.8.130"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+    _cmake = None
+
+    def configure(self):
+        self.options["aws-sdk-cpp"].monitoring = True
+        self.options["aws-sdk-cpp"].shared = True
+        self.options["aws-libfabric"].shared = True
```

Comment:
These can be overridden by consumers so if these are requirements then they should be moved to a validate method